model/pdata/metrics.go

```diff
@@ -305,3 +305,39 @@ const (
 func (at AggregationTemporality) String() string {
 	return otlpmetrics.AggregationTemporality(at).String()
 }
+
+// NumberDataPointType specifies the type of NumberDataPoint.
+type NumberDataPointType int32
+
+const (
+	NumberDataPointTypeNull NumberDataPointType = iota
```

Comment:
s/NumberDataPointTypeNull/NumberDataPointTypeNone?

Reply:
Could go w/ None, I followed the naming convention used by AttributeValue `AttributeValueTypeNull`

Reply:
Let's go with None for this, makes more sense. I will think about AttributeValue